stl/inc/mutex

```diff
@@ -45,11 +45,30 @@ public:
     _Mutex_base& operator=(const _Mutex_base&) = delete;
 
     void lock() {
-        _Check_C_return(_Mtx_lock(_Mymtx()));
+        if (_Mtx_lock(_Mymtx()) != _Thrd_success) {
+            // undefined behavior, only occurs for plain mutexes (N4928 [thread.mutex.requirements.mutex.general]/6)
+            _STD _Throw_Cpp_error(_RESOURCE_DEADLOCK_WOULD_OCCUR);
+        }
+        if (_Ownership_levels() == INT_MAX) {

```

Comment:
Nitpick: We conventionally put a newline between "non-chained" `if`-statements, to make the control flow clearer. Occurs below.